src/main/java/hw21/ProductsHandler.java

```diff
@@ -0,0 +1,55 @@
+package hw21;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class ProductsHandler {
+
+    public static List<Product> findExpensiveBooks(List<Product> products) {
+        Objects.requireNonNull(products);
+        return products.stream()
+                .filter(product -> product.isBook() && product.getPrice() > 250)
+                .toList();
+    }
+
+    public static List<Product> getBookDiscount(List<Product> products) {
```

Comment:
It's not really about getting the discount, but getting the books with the discount applied. So, let's rename it  to `getBooksWithDiscount`